docs/src/developer/design/user_api.rst

```diff
@@ -0,0 +1,154 @@
+.. _designing_cmaizes_user_api:
+
+###########################
+Designing CMaize's User API
+###########################
+
+:ref:`overview_of_cmaizes_design` called for CMaize to have a functional-style
+user :term:`API` written over top of the object-oriented core. This page
+describes the design of CMaize's user API.
+
+TODO: Complete this page. Current content was originally part of overview, but
+then I decided to break it off.
+
+***********************
+User API Considerations
+***********************
+
+.. _functional_style_and_cmake_based:
+
+functional-style and cmake-based
+   Stemming from design discussions :ref:`overview_of_cmaizes_design` it was
+   decided that the user-facing API of CMaize needed to be written in
+   traditional CMake and should assume functional-style programming.
+
+Expected Workflow Considerations
+================================
+
+The next set of considerations, in order, represent the major steps we expect
+a CMaize-based build system to take. Generally speaking these
+considerations directly address what CMake should do in each
+of the :term:`build phase`\ s. That said CMake will parse the entire
+build system during the :ref:`configure_phase`, even the components applying to
+later phases.
+
+.. _project_meta_data:
+
+project meta data
+   Following from the :ref:`functional_style_and_cmake_based` consideration the
+   build system the user writes with CMaize should be pure CMake and invoked by
+   CMake. CMake requires (technically strongly suggests) that the first things
+   a user do are:
+
+   1. Set the minimum version of CMake needed.
+
+      - This is a call to CMake's
+        `cmake_minimum_required <https://tinyurl.com/3w6n75ec>`_
+        command.
+
+   2. Define the :term:`project`.
+
+      - This is a call to CMake's
+        `project <https://cmake.org/cmake/help/latest/command/project.html>`__
+        command.
+
+.. _obtain_cmaize:
+
+obtain CMaize
+   With obligatory CMake boilerplate out of the way, the user is free to start
+   using CMaize. Since CMaize is unlikely to be included with CMake
+   distributions any time soon, the first step is to obtain CMaize. The current
+   best practice for obtaining CMake modules is through
+   `FetchContent <https://tinyurl.com/yubmtj8m>`_, *i.e.*:
+
+   1. `FetchContent_Declare <https://tinyurl.com/yzxm6y2d>`_
+   2. `FetchContent_MakeAvailable <https://tinyurl.com/mtteytj7>`_
+   3. `include(cmaize) <https://tinyurl.com/p2r8xut2>`__
+
+.. _declare_build_options:
+
+declare build options
```

Comment:
I'm guessing this is where the `cmaize_options` command would come in, right?

Reply:
yeah this was the motivation for it.